### DISCUSSION TITLE
Distilling contrastive representation encoder

### DISCUSSION THREAD

**Discussion by dimitry12** [yoshitomo-matsubara/torchdistill#218] [Q&A]
`torchdistill` is an impressive framework and I am only starting to understand how to apply it to tasks outside of image classification.

One such task is contrastive representation learning. CLIP is a good example of a model from this space. Is it correct that I would need to implement a custom "trainer" executable (similar to [examples](https://github.com/yoshitomo-matsubara/torchdistill/tree/main/examples) for other tasks) when distilling representation encoder?

**Comment by yoshitomo-matsubara** [comment]
Hi @dimitry12 

If you can provide more details (e.g., how does the input and output look? Then what would be student and teacher models?), I can help you implement it.

You will not need to edit the executable script a lot. The executable script is more or less the same with other tasks, and the main differences are validation/test-related implementations and model's input and output since all the other components are generalized and handled by training box / distillation box that is defined by a yaml config file.

**Reply by dimitry12** [reply]
Good point, I didn't compare executable scripts before but I now see that they are nearly identical except for metrics/eval.

Thank you for offering help in implementing representation distillation. Let me formulate the task and decide which distillation method I want to use first. @yoshitomo-matsubara I will get back to you in this github-discussion.
